libsignal/src/main/java/org/session/libsignal/utilities/Snode.kt

```diff
@@ -19,17 +34,39 @@ class Snode(val address: String, val port: Int, val publicKeySet: KeySet?, val v
 
     data class KeySet(val ed25519Key: String, val x25519Key: String)
 
-    override fun equals(other: Any?): Boolean {
-        return if (other is Snode) {
-            address == other.address && port == other.port
-        } else {
-            false
+    override fun equals(other: Any?) = other is Snode && address == other.address && port == other.port
+    override fun hashCode(): Int = address.hashCode() xor port.hashCode()
+    override fun toString(): String = "$address:$port"
+
+    companion object {
+        private val CACHE = mutableMapOf<String, Version>()
```

Comment:
When is the `CACHE` populated? I can only see the use of `getOrElse` but that method doesn't write into the map

Reply:
Sorry I should have mentioned this early together with my comment. There are concerns with the usage of a static cache like this, you'd need to handle these:
1. Do we keep upper limit on how many items are inserted into the cache? You probably don't want to leave it open for unlimited entries given they are never cleared during the lifecycle of the app.
2. Is there a thread safety measure in place to make sure concurrent access to the cache is safe?

Reply:
It wouldn't make a lot of sense for it to be more than a 2 or 3, but it would conceivably be an attack vector to make a malicious node that iterates through countless garbage version strings, causing OOMs, so we could use an LruCache of 100 or so.

Thread Safety 😓 I guess I'll have to account for that...

Could just synchronize that one method that uses it, or use a synchronized Map, or use a ThreadLocal, and have a different instance for each thread. 🤔

Reply:
done